R/ggplotly.R

```diff
@@ -829,6 +829,22 @@ gg2list <- function(p) {
     stop("No exportable traces")
   }
   
+  # fixed coordinates: if the coordinates ratio is not NULL, then
+  # we make the size of the plot according to the specified ratio
+  # note: we set the biggest dimension to 600
+  if (!is.null(p$coordinates$ratio)) {
+    x_range <- range(built[[2]]$ranges[[1]]$x.major_source, na.rm = TRUE)
+    y_range <- range(built[[2]]$ranges[[1]]$y.major_source, na.rm = TRUE)
+    yx_ratio <- (y_range[2] - y_range[1]) / (x_range[2] - x_range[1])
+    if (yx_ratio > 1) {
+      layout$height <- 600
+      layout$width <- layout$height * (1 / p$coordinates$ratio) * (1 / yx_ratio) 
+    } else {
+      layout$width <- 600
+      layout$height <- layout$height * (1 / p$coordinates$ratio) * yx_ratio 
+    }
```

Comment:
Hard-coding the height/width is going to make for unpleasant htmlwidget experience. I think the best way to do it is pass this ratio to the HTMLwidget.resize method and impose the ratio on the height/width there. We should also provide a way to specify this ratio for non-ggplots